src/Cameras/Inputs/BaseCameraPointersInput.ts

```diff
@@ -0,0 +1,328 @@
+                    evt.preventDefault();
+                }
+            } else if (p.type === PointerEventTypes.POINTERMOVE) {
+                if (!noPreventDefault) {
+                    evt.preventDefault();
+                }
+
+                // One button down
+                if (pointA && pointB === null) {
+                    var offsetX = evt.clientX - pointA.x;
+                    var offsetY = evt.clientY - pointA.y;
+                    this.doTouch(pointA, offsetX, offsetY);
+
+                    pointA.x = evt.clientX;
+                    pointA.y = evt.clientY;
+                }
+                // Two buttons down: pinch
+                else if (pointA && pointB) {
+                    var ed = (pointA.pointerId === evt.pointerId) ? pointA : pointB;
+                    ed.x = evt.clientX;
+                    ed.y = evt.clientY;
+                    var distX = pointA.x - pointB.x;
+                    var distY = pointA.y - pointB.y;
+                    var pinchSquaredDistance = (distX * distX) + (distY * distY);
+                    var multiTouchPanPosition = {x: (pointA.x + pointB.x) / 2,
+                                                 y: (pointA.y + pointB.y) / 2,
+                                                 pointerId: evt.pointerId,
+                                                 type: p.type};
+
+                    if (previousPinchSquaredDistance === 0) {
+                        previousPinchSquaredDistance = pinchSquaredDistance;
+                        previousMultiTouchPanPosition = {x: (pointA.x + pointB.x) / 2,
+                                                         y: (pointA.y + pointB.y) / 2,
+                                                         pointerId: evt.pointerId,
+                                                         type: p.type};
+                        return;
+                    }
+
+                    this.doMultiTouch(
+                      pointA,
+                      pointB,
+                      previousPinchSquaredDistance,
+                      pinchSquaredDistance,
+                      previousMultiTouchPanPosition,
+                      multiTouchPanPosition);
+
+                    previousMultiTouchPanPosition = {x: (pointA.x + pointB.x) / 2,
+                                                     y: (pointA.y + pointB.y) / 2,
+                                                     pointerId: evt.pointerId,
+                                                     type: p.type};
+                    previousPinchSquaredDistance = pinchSquaredDistance;
+                }
+            }
+        };
+
+        this._observer = this.camera.getScene().onPointerObservable.add(
+            this._pointerInput,
+            PointerEventTypes.POINTERDOWN | PointerEventTypes.POINTERUP |
+            PointerEventTypes.POINTERMOVE);
+
+        this._onLostFocus = () => {
+            this.log("FollowCameraPointersInput._onLostFocus");
```

Comment:
this has to be removed

Reply:
agreed.
the switchable `log()` method  i have is more complicated than it has to be.